### PR TITLE
[Enhancement] Fs starlet support batch delete api

### PR DESCRIPTION
**Pull request by TszKitLo40**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 16 / 19 (84.21%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/fs/fs.h|0|2|00.00%|[274, 275]|
|:large_blue_circle:|src/fs/fs_starlet.cpp|16|17|94.12%|[516]|

File: be/src/fs/fs_starlet.cpp [hunk 1]
```diff
@@ -504,6 +504,29 @@ class StarletFileSystem : public FileSystem {
...
+        parsed_paths.resize(paths.size());
+        std::shared_ptr<staros::starlet::fslib::FileSystem> fs = nullptr;
+        for (size_t i = 0; i < paths.size(); ++i) {
+            ASSIGN_OR_RETURN(auto pair, parse_starlet_uri(paths[i]));
+            if (i == 0) {
+                auto fs_st = get_shard_filesystem(pair.second);
+                if (!fs_st.ok()) {
+                    return to_status(fs_st.status());
+                }
+                fs = *fs_st;
```
Comment (kevincai) [inline be/src/fs/fs_starlet.cpp:518]: shouldn't report error if in case get different fs from `get_shard_filesystem` ?

File: be/src/fs/fs.h [hunk 2]
```diff
@@ -267,6 +267,13 @@ class FileSystem {
     // Given the path to a remote file, delete the file's cache on the local file system, if any.
     // On success, Status::OK is returned. If there is no cache, Status::NotFound is returned.
     virtual Status drop_local_cache(const std::string& path) { return Status::NotFound(path); }
+
+    // Batch delete the given files.
+    // return ok if all success (not found error ignored), error if any failed and the message indicates the fail message
+    // possibly stop at the first error if is simulating batch deletes.
+    virtual Status delete_files(const std::vector<std::string_view>& paths) {
+        return Status::NotSupported("FileSystem::delete_files()");
```
Comment (sduzh) [inline be/src/fs/fs.h:275]: Could you please give a default implementation.

File: be/src/fs/fs.h [hunk 3]
```diff
@@ -267,6 +267,16 @@ class FileSystem {
     // Given the path to a remote file, delete the file's cache on the local file system, if any.
     // On success, Status::OK is returned. If there is no cache, Status::NotFound is returned.
     virtual Status drop_local_cache(const std::string& path) { return Status::NotFound(path); }
+
+    // Batch delete the given files.
+    // return ok if all success (not found error ignored), error if any failed and the message indicates the fail message
+    // possibly stop at the first error if is simulating batch deletes.
+    virtual Status delete_files(const std::vector<std::string_view>& paths) {
+        for (auto path : paths) {
+            RETURN_IF_ERROR(delete_file(std::string(path)));
```
Comment (sduzh) [inline be/src/fs/fs.h:276]: Should ignore the NotFound error returned by `delete_file`

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 20 / 27 (74.07%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/fs/fs.h|0|6|00.00%|[274, 275, 276, 277, 278, 281]|
|:large_blue_circle:|src/fs/fs_starlet.cpp|20|21|95.24%|[516]|

File: be/src/fs/fs_starlet.cpp [hunk 4]
```diff
@@ -504,6 +504,34 @@ class StarletFileSystem : public FileSystem {
         return to_status((*fs_st)->drop_cache(pair.first));
     }
 
+    Status delete_files(const std::vector<std::string_view>& paths) override {
+        std::vector<std::string> parsed_paths;
+        parsed_paths.resize(paths.size());
```
Comment (sduzh) [inline be/src/fs/fs_starlet.cpp:509]:
```suggestion
        parsed_paths.reserve(paths.size());
```

File: be/src/fs/fs_starlet.cpp [hunk 5]
```diff
@@ -504,6 +504,34 @@ class StarletFileSystem : public FileSystem {
...
+                return to_status(fs_st.status());
+            }
+            if (fs == nullptr) {
+                shard_id = pair.second;
+                fs = *fs_st;
+            }
+            if (shard_id != pair.second) {
+                return Status::InternalError("Not all paths have the same scheme");
+            }
+            parsed_paths[i] = std::string(pair.first);
```
Comment (sduzh) [inline be/src/fs/fs_starlet.cpp:525]:
```suggestion
            parsed_paths.emplace_back(std::move(pair.first));
```

File: be/src/fs/fs.h [hunk 6]
```diff
@@ -267,6 +267,19 @@ class FileSystem {
     // Given the path to a remote file, delete the file's cache on the local file system, if any.
     // On success, Status::OK is returned. If there is no cache, Status::NotFound is returned.
     virtual Status drop_local_cache(const std::string& path) { return Status::NotFound(path); }
+
+    // Batch delete the given files.
+    // return ok if all success (not found error ignored), error if any failed and the message indicates the fail message
+    // possibly stop at the first error if is simulating batch deletes.
+    virtual Status delete_files(const std::vector<std::string>& paths) {
+        for (auto path : paths) {
```
Comment (sduzh) [inline be/src/fs/fs.h:275]:
```suggestion
        for (auto&& path : paths) {
```

**Review by sduzh — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 19 / 26 (73.08%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/fs/fs.h|0|6|00.00%|[274, 275, 276, 277, 278, 281]|
|:large_blue_circle:|src/fs/fs_starlet.cpp|19|20|95.00%|[516]|

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 19 / 26 (73.08%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/fs/fs.h|0|6|00.00%|[274, 275, 276, 277, 278, 281]|
|:large_blue_circle:|src/fs/fs_starlet.cpp|19|20|95.00%|[516]|

**Comment by sduzh** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**